src/ClientRepository.php

```diff
@@ -78,75 +79,110 @@ public function activeForUser($userId)
     /**
      * Store a new client.
      *
-     * @param  int|null  $userId
-     * @param  string  $name
-     * @param  string  $redirect
-     * @param  string|null  $provider
-     * @param  bool  $personalAccess
-     * @param  bool  $password
-     * @param  bool  $confidential
-     * @return \Laravel\Passport\Client
+     * @param  string[]  $redirectUris
+     * @param  string[]  $grantTypes
      */
-    public function create($userId, $name, $redirect, $provider = null, $personalAccess = false, $password = false, $confidential = true)
-    {
-        $client = Passport::client()->forceFill([
-            'user_id' => $userId,
+    protected function create(
+        string $name,
+        array $grantTypes,
+        array $redirectUris = [],
+        ?string $provider = null,
+        bool $confidential = true,
+        ?Authenticatable $user = null
+    ): Client {
+        $client = Passport::client();
+        $columns = $client->getConnection()->getSchemaBuilder()->getColumnListing($client->getTable());
```

Comment:
How about using a config option, instead of querying the column details?

Reply:
Also thought about a new config option, but it seems to be redundant when we can easily check for existing columns?

Reply:
It would save one query. But I will leave that decision to you / the Laravel Team.

Reply:
Yeah but client creation / updating doesn't happen frequently. Lets see what maintainers decide.